x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobManager.java

```diff
@@ -658,16 +666,16 @@ public void notifyFilterChanged(MlFilter filter, Set<String> addedItems, Set<Str
                 indexJobs -> {
                     threadPool.executor(MachineLearning.UTILITY_THREAD_POOL_NAME).execute(() -> {
 
-                        List<Job> allJobs = new ArrayList<>();
-                        // Check for duplicate jobs
-                        for (Job indexJob : indexJobs) {
-                            if (clusterStateJobs.containsKey(indexJob.getId())) {
-                                logger.error("[" + indexJob.getId() + "] job configuration exists in both clusterstate and index");
-                            } else {
-                                allJobs.add(indexJob);
+                        List<Job> allJobs = new ArrayList<>(indexJobs);
+                        // Duplicate configs existing in both the clusterstate and index documents are ok
+                        // this may occur during migration of configs.
+                        // Filter the duplicates so we don't update twice for duplicated jobs
+                        for (String clusterStateJobId : clusterStateJobs.keySet()) {
+                            boolean isDuplicate = allJobs.stream().anyMatch(job -> job.getId().equals(clusterStateJobId));
```

Comment:
similar concern as before with the usage of a set vs iterating over a list everytime, but no biggie.